Amazon Alexa Reviews Analysis/Model/prediction.py

```diff
@@ -0,0 +1,27 @@
+import pandas as pd
+import Amazon_Alexa_Reviews_Analysis as A
+
+# loading our model
+
+import joblib
+classifier = joblib.load(r'C:\Users\DELL\Desktop\Kaggle+HE\Github GSSoC21\Amazing Python Scripts\Amazon Alexa Reviews Analysis\Model\AmazonAlexaReviewsAnalysis.pkl')
+
+# getting a review as input
+
+a = input('Enter review: ')
+df1 = pd.read_csv(r'C:\Users\DELL\Desktop\Kaggle+HE\Github GSSoC21\Amazing Python Scripts\Amazon Alexa Reviews Analysis\Dataset\amazon_alexa.csv')
+
+l = []
+for i in df1['verified_reviews']:
+    l.append(i)
+    
+b = None    
+for j in range(0,len(l)):
+    if l[j] == a:
+        b = classifier.predict(A.X_train[j])

```

Comment:
What is this logic? The user can give any input regardless it is present in the dataset or not. Here, you are only considering the dataset values. Change this to take new inputs from user

Reply:
I'm sorry. I spent like 3 to 4 hours yesterday trying to analyze a single review using the 'classifier', but it just wasn't working. I kept getting the same error no matter how many methods or manipulations I tried. This seemed to be the only final resolve that was working. 
I'll just try again. :(

Reply:
I think you need the same preprocessing pipelines which you used to train the dataset. Maybe the new text is not getting processed

Reply:
Did you check these new screenshots that I added? @kaustubhgupta Can something be done?

Reply:
The problem is here is that the model has been trained for 1500 features but the input provided is 2 features. This can be solved by using this:

(lines 72 to 75 in Amazon Alexa Reviews Analysis/Model/Amazon_Alexa_Reviews_Analysis.py)

cv = CountVectorizer(max_features=1500, analyzer='word', lowercase=False) 

df['verified_reviews'] = df['verified_reviews'].apply(lambda x: " ".join(x) )  # to join all words in the lists

X = cv.fit_transform(df['verified_reviews'])

You need to use count vectorizer transform function to transform input to 1500 features and then fed it to model. 

I would suggest that you create a single file for modeling and prediction.